src/HexagonOptimize.cpp

```diff
@@ -1362,21 +1379,34 @@ class EliminateInterleaves : public IRMutator {
             return true;
         }
 
+        if (const Load *load = x.as<Load>()) {
+            if (buffers.contains(load->name)) {
+                return buffers.get(load->name) != BufferState::NotInterleaved;
+            }
+        }
+
+        if (const Add *op = x.as<Add>()) {
+            return yields_interleave(op->a) || yields_interleave(op->b);
```

Comment:
Shouldn't an add yield an interleave only if both operands do too?

Reply:
The idea with this change is that as long as one of the operands yields an interleave, we can deinterleave it at ~zero cost, because we're adding one op while removing another. In the case I looked at, it was *highly* profitable to make this trade. In depthwise convolution, it enables the entire convolution to be performed deinterleaved in exchange for only a single extra deinterleave almost at compute_root.